test/e2e/e2e_test.go

```diff
@@ -118,6 +122,69 @@ func createServingCertAnnotatedService(client *kubernetes.Clientset, secretName,
 				},
 			},
 		},
+	}
+	if headless {
+		service.Spec.Selector = map[string]string{
+			owningHeadlessServiceLabelName: serviceName,
+		}
+		service.Spec.ClusterIP = v1.ClusterIPNone
+	}
+	_, err := client.CoreV1().Services(namespace).Create(context.TODO(), service, metav1.CreateOptions{})
+	return err
+}
+
+func createStatefulSet(client *kubernetes.Clientset, secretName, statefulSetName, serviceName, namespace string, numReplicas int) error {
+	const podLabelName = "pod-label"
+	podLabelValue := statefulSetName + "-pod-label"
+	replicasInt32 := int32(numReplicas)
+	_, err := client.AppsV1().StatefulSets(namespace).Create(context.TODO(), &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: statefulSetName,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: &replicasInt32,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{podLabelName: podLabelValue},
+			},
+			ServiceName:         serviceName,
+			PodManagementPolicy: appsv1.ParallelPodManagement, // We want changes to happen fast, there isn't really state to maintain.
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						podLabelName:                   podLabelValue,
+						owningHeadlessServiceLabelName: serviceName,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Name:  statefulSetName + "-container",
+						Image: "docker.io/nginxinc/nginx-unprivileged:1.18-alpine",
```

Comment:
This image was chosen because I found it in https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/nginx-unprivileged_is.yaml . Is that relevant, and if so, what’s the right way to refer to it? IIRC(?) just using the ImageStream name didn’t work and e.g. https://docs.openshift.com/container-platform/3.11/dev_guide/managing_images.html#using-is-with-k8s says
> When using this feature, it is only possible to reference image streams that reside in the same project as the resource. 

Or is that set of ImageStreams not helpful?

Reply:
Never mind, I read the CI image documentation a bit more carefully.

We need a HTTPS server, of any kind. `openssl s_server` would be fine as well, sadly `/usr/bin/openssl` is not a default part of major container images, and neither is `nc`.

For now I will move to the `nicolaka/netshoot:latest` image (still from `docker.io`) which is already used in the tests (although that might not help all that much if the pods are scheduled to different nodes).

A possible alternative would be to write a ~10-line server in Python, which _is_ included at least in RHEL 7 images.

(Or maybe the full e2e test of connections might not be worth the trouble. Up to you.)